backend/src/main/java/com/bakdata/conquery/models/query/queryplan/aggregators/specific/EventDateUnionAggregator.java

```diff
@@ -0,0 +1,72 @@
+package com.bakdata.conquery.models.query.queryplan.aggregators.specific;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.bakdata.conquery.models.common.CDateSet;
+import com.bakdata.conquery.models.datasets.Column;
+import com.bakdata.conquery.models.datasets.Table;
+import com.bakdata.conquery.models.events.Bucket;
+import com.bakdata.conquery.models.externalservice.ResultType;
+import com.bakdata.conquery.models.identifiable.ids.specific.TableId;
+import com.bakdata.conquery.models.query.QueryExecutionContext;
+import com.bakdata.conquery.models.query.queryplan.aggregators.Aggregator;
+import com.bakdata.conquery.models.query.queryplan.clone.CloneContext;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Collects the event dates of all events that are applicable to the specific
+ * part of a query. Eventually the set of collected dates is tailored to the
+ * provided date restriction.
+ *
+ */
+@RequiredArgsConstructor
+public class EventDateUnionAggregator implements Aggregator<String> {
+
+	private final Set<TableId> requiredTables;
+	private Column validityDateColumn;
+	private CDateSet set = CDateSet.create();
+	private CDateSet dateRestriction;
+
+	@Override
+	public void collectRequiredTables(Set<TableId> requiredTables) {
+		requiredTables.addAll(this.requiredTables);
+	}
+
+	@Override
+	public void nextTable(QueryExecutionContext ctx, Table currentTable) {
+		validityDateColumn = Objects.requireNonNull(ctx.getValidityDateColumn());
+		if (!validityDateColumn.getType().isDateCompatible()) {
```

Comment:
da würde ich mich ehrlich gesagt eher drauf verlassen als das jedes mal zu prüfen.

Reply:
Ich hatte das in einem anderen Aggregator gesehen.